cpp/include/cudf/detail/aggregation.hpp

```diff
@@ -50,6 +52,117 @@ struct quantile_aggregation : aggregation {
   experimental::interpolation _interpolation;  ///< Desired interpolation
 };
 
+/**
+ * @brief Maps an `aggregation::Kind` value to it's corresponding binary
+ * operator.
+ *
+ * @note Not all values of `aggregation::Kind` have a valid corresponding binary
+ * operator. For these values `E`,
+ * `std::is_same_v<corresponding_operator<E>::type, void>`.
+ *
+ * @tparam k The `aggregation::Kind` value to map to its corresponding operator
+ */
+template <aggregation::Kind k> struct corresponding_operator { using type = void; };
+template <> struct corresponding_operator<aggregation::MIN> { using type = DeviceMin; };
+template <> struct corresponding_operator<aggregation::MAX> { using type = DeviceMax; };
+template <> struct corresponding_operator<aggregation::SUM> { using type = DeviceSum; };
+template <> struct corresponding_operator<aggregation::COUNT> { using type = DeviceSum; };
+template <aggregation::Kind k>
+using corresponding_operator_t = typename corresponding_operator<k>::type;
+
+/**---------------------------------------------------------------------------*
+ * @brief Determines accumulator type based on input type and aggregation.
+ *
+ * @tparam SourceType The type on which the aggregation is computed
+ * @tparam k The aggregation performed
+ *---------------------------------------------------------------------------**/
+template <typename SourceType, aggregation::Kind k, typename Enable = void>
+struct target_type_impl { using type = void; };
+
+// Computing MIN of SourceType, use SourceType accumulator
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::MIN> { using type = SourceType; };
+
+// Computing MAX of SourceType, use SourceType accumulator
+template <typename SourceType>
+struct target_type_impl<SourceType, aggregation::MAX> { using type = SourceType; };
+
+// Always use int64_t accumulator for COUNT
```

Comment:
size_type is int32m, not int64.
```suggestion
// Always use size_type accumulator for COUNT
```